src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers/WebTypeReplacements.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers
+{
+    public class WebTypeReplacements : AdditionalText
```

Comment:
This seems kind of weird. Why not add it as as an actual additional document? If there's a good reason to do it this way, maybe add a comment as to the rationale.

We should have a props file that gets packaged with the analyzers that set things as additional files and then the project will automatically get them. We may need to update UA to ensure that's loaded correctly if we're not adding an analyzer to the actualy csproj.

Reply:
Yea, I'm doing this because UA loads the analyzers directly (rather than via a csproj reference), and the AdditionalTexts are added programatically. 

I'll see about changing it to use a file on-disk. I'm not sure what the best way is to make sure the typemap file is packaged with the analyzers and then made available to the project file, but I'm sure there's a way.

Reply:
Ok, I have an update that loads type maps from on-disk type map configuration. It's a few more hoops to jump through for UA, but I think it's better overall because the same mapping files will work for UA and for the analyzers used as a standalone NuGet package.